msal/authority.py

```diff
@@ -65,7 +65,7 @@ def user_realm_discovery(self, username):
 
 def canonicalize(url):
     # Returns (canonicalized_url, netloc, tenant). Raises ValueError on errors.
-    match_object = re.match("https://([^/]+)/([^/\?#]+)", url.lower())
+    match_object = re.match(r'https://([^/]+)/([^/?#\\]+)', url.lower())
```

Comment:
Thanks very much for the catch! In such case, I think we can drop the `\\` part because it was not the intention.

By the way, can you share a link to that pytest warning message? Would like to see it in action. :-)

Reply:
No problem! Here's the message I got:

```
========================================================================================================================================================================= warnings summary =========================================================================================================================================================================
/home/marstr/.local/share/virtualenvs/microsoft-authentication-extensions-for-python/lib/python3.6/site-packages/msal/authority.py:68
  /home/marstr/.local/share/virtualenvs/microsoft-authentication-extensions-for-python/lib/python3.6/site-packages/msal/authority.py:68: DeprecationWarning: invalid escape sequence \?
    match_object = re.match("https://([^/]+)/([^/\?#]+)", url.lower())

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================================================================================================================================= 1 passed, 1 skipped, 1 warnings in 0.52 seconds ==========================================================================================================================================================
```

Reply:
Fixed in ce450a7